test/extended/builds/pipeline.go

```diff
@@ -820,4 +835,208 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
+							return false, nil
+						}
+					}
+				}
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying that the correct builds and jobs exist")
+			for buildName, buildInfo := range buildNameToBuildInfoMap {
+				_, status, err := j.GetResource(buildInfo.jenkinsBuildURI)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				if buildInfo.number%2 == 0 {
+					_, err := oc.BuildClient().Build().Builds(oc.Namespace()).Get(buildName, metav1.GetOptions{})
+					o.Expect(err).To(o.HaveOccurred())
+					o.Expect(status != http.StatusOK).To(o.BeTrue(), "Jenkins job run exists for %s but shouldn't.", buildName)
+				} else {
+					_, err := oc.BuildClient().Build().Builds(oc.Namespace()).Get(buildName, metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(status == http.StatusOK).To(o.BeTrue(), "Jenkins job run does not exist for %s but should.", buildName)
+				}
+			}
+		})
+	})
+	g.Context("delete jenkins job runs when jenkins re-establishes communications", func() {
+		g.BeforeEach(func() {
+			setupJenkins(jenkinsPersistentTemplatePath)
+
+			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
+				ticker = jenkins.StartJenkinsMemoryTracking(oc, oc.Namespace())
+			}
+
+			g.By("waiting for builder service account")
+			err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			if g.CurrentGinkgoTestDescription().Failed {
+				exutil.DumpPodStates(oc)
+				exutil.DumpPodLogsStartingWith("", oc)
+			}
+			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
+				ticker.Stop()
+			}
+		})
+
+		g.It("should delete job runs when the associated build is deleted - jenkins unreachable", func() {
+			defer exutil.CleanUp(oc)
+
+			type buildInfo struct {
+				number          int
+				jenkinsBuildURI string
+			}
+			buildNameToBuildInfoMap := make(map[string]buildInfo)
+
+			g.By(fmt.Sprintf("calling oc create -f %q", envVarsPipelinePath))
+			err := oc.Run("new-app").Args("-f", envVarsPipelinePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting 5 pipeline builds")
+			for i := 1; i <= 5; i++ {
+				// start the build
+				br, _ := exutil.StartBuildAndWait(oc, "sample-pipeline-withenvs")
+				br.AssertSuccess()
+
+				// get the build information
+				build, err := oc.BuildClient().Build().Builds(oc.Namespace()).Get(fmt.Sprintf("sample-pipeline-withenvs-%d", i), metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				jenkinsBuildURI, err := url.Parse(build.Annotations[buildapi.BuildJenkinsBuildURIAnnotation])
+
+				if err != nil {
+					fmt.Fprintf(g.GinkgoWriter, "error parsing build uri: %s", err)
+				}
+				buildNameToBuildInfoMap[build.Name] = buildInfo{number: i, jenkinsBuildURI: jenkinsBuildURI.Path}
+			}
+
+			g.By("scaling down jenkins")
+			_, err = oc.Run("scale").Args("dc/jenkins", "--replicas=0").Output()
```

Comment:
i *think* if you use an emptydir volume and the container gets restarted (such as by killing the jenkins process inside the container), the volume content gets reused/retained.

that could be a way to avoid using PVs.  However you'll have a timing challenge to delete the builds while jenkins is restarting.